expression/builtin_time.go

```diff
@@ -6120,14 +6120,90 @@ func (b *builtinTimestampAddSig) Clone() builtinFunc {
 	return newSig
 }
 
+var (
+	minDatetimeInGoTime, _ = types.MinDatetime.GoTime(time.Local)
+	minDatetimeNanos       = float64(minDatetimeInGoTime.Unix())*1e9 + float64(minDatetimeInGoTime.Nanosecond())
+	maxDatetimeInGoTime, _ = types.MaxDatetime.GoTime(time.Local)
+	maxDatetimeNanos       = float64(maxDatetimeInGoTime.Unix())*1e9 + float64(maxDatetimeInGoTime.Nanosecond())
+	minDatetimeMonths      = float64(1000*12 + 0)
```

Comment:
Why `minDatetimeMonths` starts from 12000?

Reply:
oh, that is not correct, it should be the months of 0001-01-01, which is 12 months since year 0000.

Reply:
done